tests/unit/api/config-test.js

```diff
@@ -32,3 +32,52 @@ test('it doesn\'t throw when all required params are specified', function (asser
 
   assert.ok(config);
 });
+
+test('it should convert myShopifyDomain to domain', function (assert) {
+  assert.expect(1);
+
+  const config = new Config({
+    myShopifyDomain: 'krundle',
+    apiKey: 123,
+    appId: 6
+  });
+
+  assert.equal(config.domain, 'krundle.myshopify.com', 'domain should be myshopify.com domain');
+});
+
+test('it output a deprecation warning when using myShopifyDomain', function (assert) {
+  assert.expect(4);
+
+  /* eslint-disable no-console */
+  const oldLog = console.warn;
+  let output = [];
+
+  console.warn = function () {
+    output = Array.prototype.slice.call(arguments);
+  };
+
+  new Config({
+    myShopifyDomain: 'krundle',
+    apiKey: 123,
+    appId: 6
+  });
+
+  assert.equal(output.length, 3, 'logging should have a tag for config');
+  assert.equal(output[0], '[JS-BUY-SDK]: ', 'the deprecation warning should be tagged');
+  assert.equal(output[1], 'Config - ', 'the deprecation warning should say it\'s from config');
+  assert.notEqual(output[2].indexOf('deprecated'), -1, 'the deprecation warning should describe the problem');
+  console.warn = oldLog;
+  /* eslint-enable no-console */
+});
+
+test('it should set domain', function (assert) {
+  assert.expect(1);
+
+  const config = new Config({
+    domain: 'krundle.com',
+    apiKey: 123,
+    appId: 6
+  });
+
+  assert.equal(config.domain, 'krundle.com', 'domain should be krundle.com domain');
+});
```

Comment:
This may be asking a lot, but do you think it's worthwhile testing that the deprecation pattern works? Like, take the config class, add a generic deprecated property, and add a generic transform, then verify that the deprecation loop thingy works? This may be hard to test, but maybe it's a good thing to do?